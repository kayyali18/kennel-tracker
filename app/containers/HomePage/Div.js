import styled from 'styled-components'

const Div = styled.div`
  display: inline-block;
  width: 250px;
  background: #e7b7d3;
  border: 2px dashed black;
  justify-content: space-between;
  padding: 10px;
  margin: 10px;
`
export default Div
