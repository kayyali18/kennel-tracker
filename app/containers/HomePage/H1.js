import styled from 'styled-components'

const H1 = styled.h1`
  text-align: center;
  font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
  color: #595959;
  font-size: 50px;
  font-weigth: 800;
`
export default H1
