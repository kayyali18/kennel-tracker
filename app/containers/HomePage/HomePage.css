.run-info {
  display: inline-block;
  width: 250px;
  border: 2px dashed #595959;
  justify-content: space-between;
  padding: 10px;
  margin: 10px;
}

.pet {
  text-align: center;
  font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
  color: #595959;
}
