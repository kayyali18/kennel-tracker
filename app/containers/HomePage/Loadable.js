/**
 * Asynchronously loads the component for HomePage
 */
import React from 'react'
import loadable from 'loadable-components'
import styled from 'styled-components'

import Loading from 'components/Loading'

const LoadingWrapper = styled.div`
  min-height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
`

export default loadable(() => import('./index'), {
  LoadingComponent: () => (
    <LoadingWrapper>
      <Loading />
    </LoadingWrapper>
  ),
})
