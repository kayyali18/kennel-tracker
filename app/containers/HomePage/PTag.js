import styled from 'styled-components'

const PTag = styled.div`
  text-align: center;
  font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
  color: #595959;
`
export default PTag
