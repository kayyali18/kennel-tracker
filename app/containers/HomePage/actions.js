/*
 * Home Actions
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 */

import { DISPLAY_CURR_DOGS, RUN_CURR_DOGS_SAGA } from './constants'

/**
 * Changes the input field of the form
 *
 * @param  {object} runInfo The new info to render
 *
 * @return {object}    An action object with a type of DISPLAY_CURR_DOGS
 */
export function displayCurrDogs(runInfo) {
  return {
    type: DISPLAY_CURR_DOGS,
    runInfo,
  }
}

/**
 * Fires off saga watch for this action type
 */
export function runInfoSagaWatcher() {
  return {
    type: RUN_CURR_DOGS_SAGA,
  }
}
