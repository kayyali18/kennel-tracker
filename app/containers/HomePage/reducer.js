/*
 * HomeReducer
 *
 * The reducer takes care of our data. Using actions, we can change our
 * application state.
 * To add a new action, add it to the switch statement in the reducer function
 *
 * Example:
 * case YOUR_ACTION_CONSTANT:
 *   return state.set('yourStateVariable', true);
 */
import { fromJS } from 'immutable'

import { DISPLAY_CURR_DOGS } from './constants'

// The initial state of the App
export const initialState = fromJS({
  runInfo: {},
})

function homeReducer(state = initialState, action) {
  switch (action.type) {
    case DISPLAY_CURR_DOGS:
      return state.set('runInfo', action.runInfo)
    default:
      return state
  }
}

export default homeReducer
