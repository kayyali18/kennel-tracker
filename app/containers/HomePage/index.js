/*
 * HomePage
 *
 * This is the first thing users see of our App, at the '/' route
 *
 * NOTE: while this component should technically be a stateless functional
 * component (SFC), hot reloading does not currently support SFCs. If hot
 * reloading is not a necessity for you then you can refactor it and remove
 * the linting exception.
 */

import React from 'react'
import PropTypes from 'prop-types'
// import { FormattedMessage } from 'react-intl'
import { Helmet } from 'react-helmet'
import { connect } from 'react-redux'
import { compose } from 'redux'
import { createStructuredSelector } from 'reselect'

import { DAEMON } from 'utils/constants'
import injectReducer from 'utils/injectReducer'
import injectSaga from 'utils/injectSaga'
import { makeSelectLoading, makeSelectError } from 'containers/App/selectors'
import BookingForm from 'components/BookingForm'
import Loading from 'components/Loading'
import Wrapper from './Wrapper'
import { makeSelectRunInfo } from './selectors'
// import messages from './messages'
import reducer from './reducer'
import saga from './saga'
import { runInfoSagaWatcher } from './actions'
import './HomePage.css'

// import DogRun from '../DogRunPage'

/* eslint-disable react/prefer-stateless-function */
export class HomePage extends React.PureComponent {
  async componentDidMount() {
    const { dispatchSaga } = this.props
    await dispatchSaga()
    setTimeout(() => {
      dispatchSaga()
    }, 0)
  }

  render() {
    const { data } = this.props
    let dogInfo
    try {
      dogInfo = data.data.map(dog => (
        <div className="run-info">
          <p className="pet">Run Number: {dog.attributes.runNumber}</p>
          <p className="pet">
            Pet Name: {dog.attributes.pet.data.attributes.name}
          </p>
          <p className="pet">
            Species: {dog.attributes.pet.data.attributes.species}
          </p>
          <p className="pet">
            Breed: {dog.attributes.pet.data.attributes.breed}
          </p>
        </div>
      ))
    } catch {
      console.log('error')
    }

    if (!data) {
      return <Loading />
    }

    return (
      <div>
        <Helmet>
          <title>Home Page</title>
          <meta
            name="HomePage with relevant info"
            content="Content explaining current dogs in the Kennel"
          />
        </Helmet>
        <Wrapper>
          <div>{dogInfo}</div>

          <BookingForm />
        </Wrapper>
      </div>
    )
  }
}

export const mapDispatchToProps = dispatch => ({
  dispatchSaga: () => dispatch(runInfoSagaWatcher()),
})

const mapStateToProps = createStructuredSelector({
  loading: makeSelectLoading(),
  error: makeSelectError(),
  data: makeSelectRunInfo(),
})

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
)

const withReducer = injectReducer({ key: 'home', reducer })
const withSaga = injectSaga({ key: 'home', saga, mode: DAEMON })

export default compose(
  withReducer,
  withSaga,
  withConnect,
)(HomePage)

HomePage.propTypes = {
  // loading: PropTypes.bool,
  // error: PropTypes.oneOfType([PropTypes.object, PropTypes.bool]),
  data: PropTypes.oneOfType([PropTypes.object, PropTypes.bool]),
  dispatchSaga: PropTypes.func,
}
