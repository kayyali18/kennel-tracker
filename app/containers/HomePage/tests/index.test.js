/* eslint-disable */

import React from 'react'
import { FormattedMessage } from 'react-intl'
import { shallow } from 'enzyme'
// import renderer from 'react-test-renderer'

import HomePage from '../index'
import messages from '../messages'

describe('<HomePage />', () => {
  it.skip('should render the page message', () => {
    const renderedComponent = shallow(<HomePage />)
    expect(
      renderedComponent.contains(<FormattedMessage {...messages.header} />),
    ).toEqual(true)
  })
  // it('should match snapshot', () => {
  //   const renderedComponent = renderer.create(<HomePage />).toJSON()
  //   expect(renderedComponent).toMatchSnapshot()
  // })
})
