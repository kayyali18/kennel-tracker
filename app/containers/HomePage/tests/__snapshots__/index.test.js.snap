// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should match snapshot 1`] = `<withReducer(withSaga(Connect(HomePage))) />`;
